### PR TITLE
Feat applications list

### PR DESCRIPTION
**Pull request by mixcas**
- Adds migrations so applications now have `userEmail` and `galleryName`. This is to make filtering easier.
- Add component to modify applications subscription. Changes `search` for text search on gallery name and user email, `status` to filtre bty application status (in process, completed, submitted and paid), `orderBy` to sort the applications by gallery name (asc/desc) and user email (asc/desc)
- Add code to ensure that `userEmail` is added to the application when creating a new application in the future.
- Applications pagination now works thru the query param `page`. Ex `?page=2`. Pagination buttons got fixed to work with this.

**NEEDS STYLING, preferably in this branch**

### CODE REVIEW

File: app/imports/components/applications/applicationsLayout.jsx [hunk 1]
```diff
@@ -40,6 +40,7 @@ export class ApplicationsLayout extends Component {
     return (
       <section id="applications" className="fluid-col s-12 m-12">
         <h3><T>applications.title</T></h3>
+        <ApplicationsControls />
```
Comment (varffs) [inline app/imports/components/applications/applicationsLayout.jsx]: could be a better name maybe? Controls more like Filters

File: app/imports/containers/applicationsContainer.jsx [hunk 2]
```diff
@@ -7,15 +7,16 @@ import { ApplicationsLayout } from '/imports/components/applications/application
 import { Applications } from '/imports/collections/applications.js';
 
 const composer = (props, onData) => {
+  console.log(props);
```
Comment (varffs) [inline app/imports/containers/applicationsContainer.jsx]: oops

File: app/imports/containers/applicationsContainer.jsx [hunk 3]
```diff
@@ -7,15 +7,16 @@ import { ApplicationsLayout } from '/imports/components/applications/application
...
+  let status = props.queryParams.status || null;
+  let search = props.queryParams.search || null;
+  let sortBy = props.queryParams.sortBy || null;
 
-  if (props.page) {
-    page = props.page;
-  }
-
-  const subscription = Meteor.subscribe('applications.index', posts, page);
+  // TODO: send the filters here
```
Comment (varffs) [inline app/imports/containers/applicationsContainer.jsx]: no longer todo?

File: app/imports/startup/server/publications.js [hunk 4]
```diff
@@ -128,11 +128,68 @@ Meteor.publishComposite('applications.index', function(posts, page) {
 
   let skip = posts * page;
 
+  let query = {};
+
+  let projection = {
```
Comment (varffs) [inline app/imports/startup/server/publications.js:133]: why `projection` as the name?

**Review by varffs — COMMENTED** [review]
all seems fine just minor questions

File: app/imports/startup/server/publications.js (hunk 4 above)
Comment (mixcas) [reply app/imports/startup/server/publications.js:133]: thats what it is 
https://docs.mongodb.com/manual/reference/operator/query/

File: app/imports/components/applications/applicationsLayout.jsx (hunk 1 above)
Comment (mixcas) [reply app/imports/components/applications/applicationsLayout.jsx]: kk, will change to `ApplicationsFilters`

File: app/imports/containers/applicationsContainer.jsx (hunk 2 above)
Comment (mixcas) [reply app/imports/containers/applicationsContainer.jsx]: will remove

File: app/imports/containers/applicationsContainer.jsx (hunk 3 above)
Comment (mixcas) [reply app/imports/containers/applicationsContainer.jsx]: will remove

**Comment by mixcas** [comment]
Ready @varffs

**Comment by mixcas** [comment]
Who will style this? @mikeyrayvon @varffs

**Comment by mikeyrayvon** [comment]
i can do it

**Comment by varffs** [comment]
It should be possible to hide the next button in a view that has less than the pagination number of posts. For example when filtering to a status that has no posts there is still a next button. The instance where we can't tell if there is a next button should only when the total number of posts perfectly divides by the posts per page

**Comment by varffs** [comment]
But that could just be called a bug as it would be nice to push this

**Comment by mixcas** [comment]
I think in the container is possible to check if there's actually another
page of posts and pass that as a parameter

**Pull request merged**